presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -518,7 +518,7 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
                         .forEach(clause -> map(clause.getRight(), clause.getLeft()));
             }
 
-            return new JoinNode(node.getId(), node.getType(), left, right, canonicalCriteria, canonicalFilter, canonicalLeftHashSymbol, canonicalRightHashSymbol);
+            return new JoinNode(node.getId(), node.getType(), left, right, canonicalCriteria, canonicalizeAndDistinct(node.getOutputSymbols()), canonicalFilter, canonicalLeftHashSymbol, canonicalRightHashSymbol);
```

Comment:
Why wouldn't that work? canonicalizeAndDistinct takes the list of symbols, canonicalizes them and removes duplicates while preserving the relative order. If the original node was a cross join, this should result in, effectively, the left child's outputs followed by the right child's outputs.